src/ast/mod.rs

```diff
@@ -2241,7 +2241,14 @@ impl fmt::Display for Statement {
                 if !after_columns.is_empty() {
                     write!(f, "({}) ", display_comma_separated(after_columns))?;
                 }
-                write!(f, "{source}")?;
+
+                if let Some(source) = source {
+                    write!(f, "{source}")?;
+                }
+
+                if source.is_none() && columns.is_empty() {
```

Comment:
```suggestion
                if source.is_none() {
```

I think we should never print `INSERT INTO my_table(my_column);`, which is syntactically invalid.

`INSERT INTO my_table(my_column) DEFAULT VALUES;` should successfully round-trip through sqlparser, since it's a logical error, but not a syntax error.

Reply:
Thanks for the review @lovasoa.

>INSERT INTO my_table(my_column) DEFAULT VALUES; should successfully round-trip through sqlparser, since it's a logical error, but not a syntax error.

I thought about this a bit and double checked the [SQL grammar](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#from-default) and [implementation in Postgres](https://github.com/postgres/postgres/blob/0bc726d95a305ca923b1f284159f40f0d5cf5725/src/backend/parser/gram.y#L12090-L12095) and both seem to make `INSERT INTO my_table(my_column) DEFAULT VALUES;` invalid at the syntax level. I think it'd make sense for sqlparser to make this scenario a syntax error as well, but I could be missing something.

I also tested a statement out with Postgres in a [DB fiddle](https://www.db-fiddle.com/f/oiiMsDZ5g8727mnEvEx1Px/0) and confirmed that it raises a syntax error:
```sql
-- error: syntax error at or near "DEFAULT"
INSERT INTO test_table (test_column) DEFAULT VALUES;
```

What do you think?

Reply:
You are right about postgres, but it's not a syntax error in sqlite. 

And `insert into t(x);` is a syntax error everywhere anyway, so I think we should never emit that.

Reply:
Ah interesting, I was not aware of the behaviour in SQLite.

I added some coverage for syntax errors with `DEFAULT VALUES` [here](https://github.com/sqlparser-rs/sqlparser-rs/pull/1036/files#diff-ae2dab2c924f4018c1daf0743b8fd912c44c5b63ae7144dd8b951bb3bd254349R186-R214). Do you think that that behaviour looks reasonable?